source/adapters/level_zero/usm.cpp

```diff
@@ -829,6 +884,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMPoolCreate(
   try {
     *Pool = reinterpret_cast<ur_usm_pool_handle_t>(
         new ur_usm_pool_handle_t_(Context, PoolDesc));
+
+    std::shared_lock<ur_shared_mutex> ContextLock(Context->Mutex);
+    Context->UsmPoolHandles.push_back(*Pool);
```

Comment:
`Context->UsmPoolHandles` should also be updated in `urUSMPoolRelease`.

Reply:
thanks @kswiecicki. missed it. fixed.